pkg/configurations/configurations.go

```diff
@@ -0,0 +1,157 @@
+package configurations
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/thoas/go-funk"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	"github.com/spf13/viper"
+)
+
+const defaultInterfacesFilter = "veth*"
+const defaultNodeNetworkStateRefresh = 5
+
+var (
+	log        = logf.Log.WithName("configurations")
+	configPath string
+)
+
+// Takes os environment values
+func init() {
```

Comment:
We can get rid of this init (I know, we use them on other places, but it is not a good practice since it randomly panics during imports).

This module should have two structs `Config`, that we already have and `ConfigWatcher`. In main, we should call `NewConfigWatcher`, initializing the structer, that would check the config path. Then we call configWatcher.Start.

Let's try to be clean and safe when dealing with singletons.